### ISSUE TITLE
[GraphQL] : Content Types with specific field names break the GraphQL API

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#17515]
Reported via Support: [99539](https://dotcms.zendesk.com/agent/tickets/99539)
Reproducible in dotCMS 5.1.6, and 5.2.0

The fields added to a Content Type in dotCMS can have almost any name, unless indicated otherwise as there are internal reserved words. However, specifically with the GraphQL API, there might be field names that map to internal dotCMS properties that break this functionality.

### Steps to reproduce the behavior
Please ask @jcastro-dotcms for the database dump to reproduce the problem:
1. Start up a dotCMS 5.1.6 with the required database backup.
2. Install Insomnia for executing GraphQL queries.
3. Run the following GraphQL query in there:
```
{
  search(query: "title:demo", limit: 10) {
    title
    modDate
    modUser {
      firstName
      lastName
    }
  }
}
```
4. Look at the `catalina.out` file to see the error message similar to this:
```
ERROR validation.InvalidSchemaException: invalid schema:
object type '<CUSTOMER-CONTENT-TYPE-1>' does not implement interface 'WidgetBaseType' because field 'host' is defined as 'SiteOrFolder!' type and not as 'Site' type
object type '<CUSTOMER-CONTENT-TYPE-1>' does not implement interface 'Contentlet' because field 'host' is defined as 'SiteOrFolder!' type and not as 'Site' type
object type '<CUSTOMER-CONTENT-TYPE-2>' does not implement interface 'WidgetBaseType' because field 'widgetPreexecute' is missing
object type '<CUSTOMER-CONTENT-TYPE-2>' does not implement interface 'WidgetBaseType' because field 'widgetTitle' is missing
object type '<CUSTOMER-CONTENT-TYPE-2>' does not implement interface 'WidgetBaseType' because field 'widgetUsage' is missing
graphql.schema.validation.InvalidSchemaException: invalid schema:
object type '<CUSTOMER-CONTENT-TYPE-1>' does not implement interface 'WidgetBaseType' because field 'host' is defined as 'SiteOrFolder!' type and not as 'Site' type
object type '<CUSTOMER-CONTENT-TYPE-1>' does not implement interface 'Contentlet' because field 'host' is defined as 'SiteOrFolder!' type and not as 'Site' type
object type '<CUSTOMER-CONTENT-TYPE-2>' does not implement interface 'WidgetBaseType' because field 'widgetPreexecute' is missing
object type '<CUSTOMER-CONTENT-TYPE-2>' does not implement interface 'WidgetBaseType' because field 'widgetTitle' is missing
object type '<CUSTOMER-CONTENT-TYPE-2>' does not implement interface 'WidgetBaseType' because field 'widgetUsage' is missing
```

### Expected behavior
The GraphQL query should work without problems.

### Additional context
As discussed with @dsilvam, some code in the `com.dotcms.graphql.InterfaceType` class needs to be updated. Regarding the error with the `<CUSTOMER-CONTENT-TYPE-2>`, the workaround is to add the missing fields to it, as indicated in the error messages.

However, for specific Content Types, an Upgrade Task should be created in order to prevent situations like these in the future.

## Acceptance Criteria

1. Unit test

**Comment by jcastro-dotcms** [comment]
PR: https://github.com/dotCMS/core/pull/17875

**Comment by dsilvam** [comment]
Please retest on master

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/17935

**Comment by dsilvam** [comment]
Note to QA:
To reproduce, create a new content type, like for example a new widget type and and a "host" field of a whatever type (categories) for example.  Before the fix, this lead to issues when generating the schema because "host" graphql field was inherited from the parent "Contentlet" graphql type but its graphql type was different than the type specified by the user when adding it to the new content type.

**Comment by nollymar** [comment]
**Passed Internal QA:** 
Tests results: Success

**Comment by dsilvam** [comment]
Needs doc: Every GraphQL type inherits from an Interface _Contentlet_ which makes a number of fields available to be used on any query. The _Contentlet_ interface includes the following GraphQL fields:

[modDate, title, titleImage, contentType, baseType, live, working, archived, locked, conLanguage, identifier, inode, host, folder, urlMap, owner, modUser]. 

If there's a variable of a Field (of a Content Type) that matches the name of any of the GraphQL fields listed, the GraphQL Schema won't be able to be generated because of type clashing. The fix in that case involves recreating the offending field with a new name so the variable doesn't clash with the GraphQL fields, or modifying the variable of the existing field via DB.

**Comment by dsilvam** [comment]
PR 2: https://github.com/dotCMS/core/pull/17963

**Comment by dsilvam** [comment]
**Needs work**: the window of clashing cases is smaller. We need to take into consideration not only the variable name of the field, but also the mapped GraphQL Type (custom type)

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/17989

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.6 // Postgres // FF

**Comment by dsilvam** [comment]
**Note to QA:** Test using [Content Type Collections](https://dotcms.com/docs/latest/graphql#Collections) or [Search + Inline Fragments](https://dotcms.com/docs/latest/graphql#InlineFragments)

**Comment by nollymar** [comment]
Tests results: Passed

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.6 // Postgres // FF 
Tested adding a new content type with fields with reserved names and then try to execute queries like those ones: 
```
{
  myTestCollection{
    title
    host1
    modDate
    modUser {
      firstName
      lastName
    }
  }
}
```
and,
```
{
  search(query: "mytest.host1:abc*", limit: 10) {
    title
    modDate
    modUser {
      firstName
      lastName
    }
  }
}
```
